source/uri-options/tests/concern-options.json

```diff
@@ -0,0 +1,78 @@
+{
+  "tests": [
+    {
+      "description": "Valid read and write concern are parsed correctly",
+      "uri": "mongodb://example.com/?readConcernLevel=majority&w=5&wTimeoutMS=30000&journal=false",
+      "valid": true,
+      "warning": false,
+      "hosts": null,
+      "auth": {
+        "username": null,
+        "password": null,
+        "db": null
+      },
+      "options": {
+        "readConcernLevel": "majority",
+        "w": "5",
+        "wTimeoutMS": "30000",
+        "journal": "false"
+      }
+    },
+    {
+      "description": "Arbitrary string readConcernLevel does not cause a warning",
+      "uri": "mongodb://example.com/?readConcernLevel=arbitraryButStillValid",
+      "valid": true,
+      "warning": false,
+      "hosts": null,
+      "auth": {
+        "username": null,
+        "password": null,
+        "db": null
+      },
+      "options": {
+        "readConcernLevel": "arbitraryButStillValid"
+      }
+    },
+    {
+      "description": "Arbitrary string w doesn't cause a warning",
+      "uri": "mongodb://example.com/?w=arbitraryButStillValid",
+      "valid": true,
+      "warning": false,
+      "hosts": null,
+      "auth": {
+        "username": null,
+        "password": null,
+        "db": null
+      },
+      "options": {
+        "w": "arbitraryButStillValid"
+      }
+    },
+    {
+      "description": "Invalid wTimeoutMS causes a warning",
```

Comment:
For wTimeoutMS, I think it's fine to leave it off to allow the C driver (and the drivers that wrap it, if they also implement this suite) to more easily run the tests.

Do you think it would be better to explicitly test for all of those things? I'm not sure if there are any other existing deviations like the C driver one for wTimeoutMS that we would want to accommodate or not.

Reply:
Testing out-of-range (at least for smaller values; not trying to test 32-bit or 64-bit limits) for numeric values seems sensible and not too difficult. I'm only aware of `wTimeoutMS` and `-1` for libmongoc, so I think the `wTimeoutMS` test could just use `-2` and document with a YAML comment that we're only doing so to account for libmongoc-based drivers supporting a legacy value.

Reply:
Testing that `-2` gives a warning seems reasonable to me, so I'll add that for each of the tests it makes sense for